subsys/bluetooth/audio/shell/audio.h

```diff
@@ -166,6 +189,66 @@ static inline void print_base(const struct shell *sh, const struct bt_bap_base *
 }
 #endif /* BROADCAST_SNK_SUBGROUP_CNT > 0 */
 
+static inline void copy_unicast_stream_preset(struct unicast_stream *stream,
+					      const struct named_lc3_preset *named_preset)
+{
+	memcpy(&stream->qos, &named_preset->preset.qos, sizeof(stream->qos));
+	memcpy(&stream->codec, &named_preset->preset.codec, sizeof(stream->codec));
+
+#if CONFIG_BT_CODEC_MAX_DATA_COUNT > 0 && CONFIG_BT_CODEC_MAX_DATA_LEN > 0
+	/* Need to update the `bt_data.data` pointer to the new value after copying the codec */
+	for (size_t i = 0U; i < ARRAY_SIZE(stream->codec.data); i++) {
+		const struct bt_codec_data *preset_data = &named_preset->preset.codec.data[i];
+		struct bt_codec_data *data = &stream->codec.data[i];
+
+		data->data.data = data->value;
+		data->data.data_len = preset_data->data.data_len;
+		memcpy(data->value, preset_data->data.data, preset_data->data.data_len);
+	}
+#endif /* CONFIG_BT_CODEC_MAX_METADATA_COUNT > 0 && CONFIG_BT_CODEC_MAX_DATA_LEN > 0 */
+
+#if CONFIG_BT_CODEC_MAX_METADATA_COUNT > 0 && CONFIG_BT_CODEC_MAX_DATA_LEN > 0
+	for (size_t i = 0U; i < ARRAY_SIZE(stream->codec.meta); i++) {
+		const struct bt_codec_data *preset_data = &named_preset->preset.codec.meta[i];
+		struct bt_codec_data *data = &stream->codec.meta[i];
+
+		data->data.data = data->value;
+		data->data.data_len = preset_data->data.data_len;
+		memcpy(data->value, preset_data->data.data, preset_data->data.data_len);
+	}
+#endif /* CONFIG_BT_CODEC_MAX_METADATA_COUNT > 0 && CONFIG_BT_CODEC_MAX_DATA_LEN > 0 */
+}
+
+static inline void copy_broadcast_source_preset(struct broadcast_source *source,
+						const struct named_lc3_preset *named_preset)
+{
+	memcpy(&source->qos, &named_preset->preset.qos, sizeof(source->qos));
+	memcpy(&source->codec, &named_preset->preset.codec, sizeof(source->codec));
+
+#if CONFIG_BT_CODEC_MAX_DATA_COUNT > 0 && CONFIG_BT_CODEC_MAX_DATA_LEN > 0
+	/* Need to update the `bt_data.data` pointer to the new value after copying the codec */
+	for (size_t i = 0U; i < ARRAY_SIZE(source->codec.data); i++) {
+		const struct bt_codec_data *preset_data = &named_preset->preset.codec.data[i];
+		struct bt_codec_data *data = &source->codec.data[i];
+
+		data->data.data = data->value;
+		data->data.data_len = preset_data->data.data_len;
+		memcpy(data->value, preset_data->data.data, preset_data->data.data_len);
```

Comment:
nitpick: If we use `preset_data->data.data_len` here twice, one line after another - maybe this will be tidier if you would create a variable `uint8_t len = preset_data->data.data_len`? There will be a bit less 'data' in those lines. Just a suggestion, I'm leaving this up to you :)